physics-informed-neural-networks-for-steady-cavity-flow/README.md

```diff
@@ -0,0 +1,274 @@
+    end
+
+    % Checkpoint models.
+    if mod(epoch, checkpointFrequency) == 0
+        fname = checkpointDirName + "/checkpoint" + epoch + ".mat";
+        save(fname, "epoch" , "net", "solverState");
+    end
+end
+```
+
+```matlabTextOutput
+Epoch=1, Loss=0.0478566, LossEqnX=0.00161311, LossEqnY=0.000700008, LossBC=0.0455435, StepTime=6.9278(sec), TotalTime=0.115463(min)
+Epoch=1000, Loss=0.0111163, LossEqnX=0.00113609, LossEqnY=0.00175239, LossBC=0.00822781, StepTime=0.549713(sec), TotalTime=10.533(min)
+Epoch=2000, Loss=0.00458981, LossEqnX=0.000568193, LossEqnY=0.000446531, LossBC=0.00357509, StepTime=0.536273(sec), TotalTime=20.0007(min)
+Epoch=3000, Loss=0.00323042, LossEqnX=0.000332672, LossEqnY=0.000292068, LossBC=0.00260568, StepTime=0.589592(sec), TotalTime=29.5363(min)
+Epoch=4000, Loss=0.00251265, LossEqnX=0.00022759, LossEqnY=0.000254729, LossBC=0.00203033, StepTime=0.55894(sec), TotalTime=39.0617(min)
+Epoch=5000, Loss=0.00196831, LossEqnX=0.000144764, LossEqnY=0.000253774, LossBC=0.00156977, StepTime=0.71958(sec), TotalTime=48.6113(min)
+Epoch=6000, Loss=0.00171982, LossEqnX=0.000118098, LossEqnY=0.000202597, LossBC=0.00139912, StepTime=0.555523(sec), TotalTime=58.1185(min)
+Epoch=7000, Loss=0.00155908, LossEqnX=0.000107693, LossEqnY=0.00017737, LossBC=0.00127401, StepTime=0.550588(sec), TotalTime=67.5455(min)
+Epoch=8000, Loss=0.00138491, LossEqnX=0.000104246, LossEqnY=0.000154897, LossBC=0.00112576, StepTime=0.618169(sec), TotalTime=77.1895(min)
+Epoch=9000, Loss=0.00125279, LossEqnX=8.98351e-05, LossEqnY=0.000136248, LossBC=0.00102671, StepTime=0.661462(sec), TotalTime=87.8465(min)
+Epoch=10000, Loss=0.00115977, LossEqnX=6.97302e-05, LossEqnY=0.000131284, LossBC=0.00095876, StepTime=0.611056(sec), TotalTime=99.4731(min)
+```
+
+# Plot predictions
+```matlab
+% Create test set using meshgrid.
+numTestSamples = 100;
+x = linspace(0, 1, numTestSamples)';
+y = x;
+[xt, yt] = meshgrid(x, y);
+
+% Flatten gridpoints and prepare data.
+xTest = dlarray(xt(:));
+yTest = dlarray(yt(:));
+if canUseGPU
+    xTest = gpuArray(xTest);
+    yTest = gpuArray(yTest);
+end
+
+% Evaluate the network.
+[psiTest, pTest, uTest, vTest] = dlfeval(@calculateStreamfunctionPressureAndVelocity, net, xTest, yTest);
+
+% Return predictions to grid and plot.
+ut = unflattenAndExtract(uTest, numTestSamples);
+vt = unflattenAndExtract(vTest, numTestSamples);
+pt = unflattenAndExtract(pTest, numTestSamples);
+psit = unflattenAndExtract(psiTest, numTestSamples);
+
+figure; 
+subplot(2,2,1)
+contourf(xt, yt, psit)
+colorbar
+axis equal
+title('psi')
+
+subplot(2,2,2)
+contourf(xt, yt, pt)
+colorbar
+axis equal
+title('p')
+
+subplot(2,2,3)
+contourf(xt, yt, ut)
+colorbar
+axis equal
+title('u')
+
+subplot(2,2,4)
+contourf(xt, yt, vt)
+colorbar
+axis equal
+title('v')
+```
+
+![figure_0.png](README_media/figure_0.png)
```

Comment:
This link is broken, should be something like `./images/figure_0.png` for the path.